File: scripts/R/get_all_device_categories_which_have_a_certain_word_in_the_name.R

library(httr)
r <- GET("https://data.oceannetworks.ca/api/deviceCategories", 
         query = list(method="get", 
                      token="YOUR_TOKEN_HERE", #>replace YOUR_TOKEN_HERE with your personal token obtained from the 'Web Services API' tab at https://data.oceannetworks.ca/Profile when logged in.
                      propertyCode="differentialtemperature"))

if (http_error(r)) {
  if (r$status_code == 400){
    error = content(r)
    str(error)
  } else {
    str(http_status(r)$message)
  }
} else {
  deviceCategories = content(r)
  for (deviceCategory in deviceCategories){
    str(deviceCategory)
  }
}

File: scripts/MATLAB/get_all_device_categories_which_have_a_certain_word_in_the_name.m

url = ['https://data.oceannetworks.ca/api/deviceCategories' ...
        '?method=get' ...
        '&token=YOUR_TOKEN_HERE' ...                          %replace YOUR_TOKEN_HERE with your personal token obtained from the 'Web Services API' tab at https://data.oceannetworks.ca/Profile when logged in.
        '&propertyCode=differentialtemperature'];

request = matlab.net.http.RequestMessage;
uri = matlab.net.URI(url);
options = matlab.net.http.HTTPOptions('ConnectTimeout',60);
  
response = send(request,uri,options);
  
if (response.StatusCode == 200)    % HTTP Status - OK
    deviceCategories = response.Body.Data;
    for i=1:numel(deviceCategories)
        deviceCategory = deviceCategories(i);
        disp(deviceCategory);
    end
elseif (response.StatusCode == 400) % HTTP Status - Bad Request
    disp(response.Body.Data.errors);
else % all other HTTP Statuses
    disp(char(response.StatusLine));
end
